budget-app/budget.py

```diff
@@ -0,0 +1,113 @@
+class Category:
+
+    def __init__(self, category):
+        self.ledger = []
+        self.amount = 0
+        self.name = category
+
+    def deposit(self, amount, desc=""):
+        self.ledger.append({"amount": amount, "description": desc})
+        self.amount += amount
+
+    def withdraw(self, amount, desc=""):
+
+        if self.check_funds(amount):
+            self.amount -= amount
+            self.ledger.append({"amount": -amount, "description": desc})
+            return True
+
+        return False
+
+    def get_balance(self):
+
+        return self.amount
+
+    def check_funds(self, amount):
+
+        return False if self.amount < amount else True
+
+    def transfer(self, amount, category):
+
+        if self.check_funds(amount):
+            self.amount -= amount
+            self.ledger.append({"amount": -amount, "description": "Transfer to " + category.name})
+            category.deposit(amount, "Transfer from " + self.name)
+            return True
+
+        return False
+
+    def __str__(self):
+
+        max_len = 30
+        header = self.name.center(max_len, "*")
+        content = ""
+
+        for item in self.ledger:
+            content += item['description'].ljust(23, " ")[:23]
+            content += "{0:>7.2f}".format(item['amount'])
+            content += "\n"
+        total = self.get_balance()
```

Comment:
No need of `total` variable .... it is the current balance and we can directly call the function where needed